tls/s2n_handshake_io.c

```diff
@@ -1052,47 +1056,53 @@ int s2n_negotiate(struct s2n_connection *conn, s2n_blocked_status * blocked)
         /* Flush any pending I/O or alert messages */
         GUARD(s2n_flush(conn, blocked));
 
-        /* If the handshake was paused, retry the current message */
-        if (conn->handshake.paused) {
-            *blocked = S2N_BLOCKED_ON_APPLICATION_INPUT;
-            GUARD(s2n_handle_retry_state(conn));
-        }
-
         if (ACTIVE_STATE(conn).writer == this) {
             *blocked = S2N_BLOCKED_ON_WRITE;
-            if (s2n_handshake_write_io(conn) < 0 && s2n_errno != S2N_ERR_BLOCKED) {
-                /* Non-retryable write error. The peer might have sent an alert. Try and read it. */
-                const int write_errno = errno;
-                const int write_s2n_errno = s2n_errno;
-                const char *write_s2n_debug_str = s2n_debug_str;
-
-                if (s2n_handshake_read_io(conn) < 0 && s2n_errno == S2N_ERR_ALERT) {
-                    /* s2n_handshake_read_io has set s2n_errno */
+            int r = s2n_handshake_write_io(conn);
```

Comment:
And this?  Seems to be declared in multiple scopes but only written once in each